### DISCUSSION TITLE
Troubleshooting MIDI in

### DISCUSSION THREAD

**Discussion by nullpainter** [copych/AcidBox#36] [Q&A] [answered]
I'm keen to use both Acid Banger as well as MIDI In so I can control Acidbox from my PC.

I've uncommented `#define MIDI_VIA_SERIAL` and have Hairless MIDI and loopMIDI installed.

I've managed to record MIDI out from Acidbox, however can't get MIDI in working. Hairless MIDI is showing that it's sending MIDI notes, but I'm not hearing anything from the DAC. The LEDs on my ESP32 S3 also blink when I'm sending MIDI notes from my PC. Acid Banger works fine. I've closed Arduino IDE so there's no serial port conflict that I can see.

Is there anything obvious that I'm missing? 

![image](https://github.com/user-attachments/assets/555c80b3-f7d5-41ef-ab72-c4b5905e1175)

**Comment by copych** [comment]
@nullpainter  Thanks for the interest! Currently it's a bit of pain to find a working HAIRLESS combination for the particular hardware. And different Arduino ESP core versions and different board definitions are only confusing even more. This is due to the fact that some boards do have an UART chip, while others do not. Usually boards with just one USB port don't have such chip.
What is your case?
* If there are two ports, one is OTG and the other is marked as COM or UART. Use UART one to connect to PC. Use `#define BOARD_HAS_UART_CHIP` in the config.h
* If your board has only one USB port, then comment this line `// #define BOARD_HAS_UART_CHIP` in the config.h
* With such changes in the config.h try to compile for your particular board name selected as a target in the tools menu. If not succeeded, then compile for the standard ESP32S3 DevBoard, having checked on the PSRAM option.

Please, let me know if this helped.

**Reply by copych** [reply, marked as answer]
Also, please, check this branch https://github.com/copych/AcidBox/tree/S3-only
as it may have some critical changes for the subject.

**Comment by nullpainter** [comment]
I don't know if was the `S3-only` branch, connecting the right USB port to the PC, or something else - but it now works! Many thanks.
